### ISSUE TITLE
Add CLI flag to force coloured log output

### ISSUE THREAD

**Issue by Sajjon** [sigp/lighthouse#3527]
## Description

I'm setting up Lighthouse to start as a background service on macOS using LaunchAgents. The LaunchAgent is a background running service which starts lighthouse by running a bash script I've written, let's call it `start_lighthouse_bn.sh`. Due to how LaunchAgents work on macOS I need to source my ENV variables first, which my script depends on, allowing me for neat start lighthouse command.

I want to `tail -f` my Lighthouse log in my terminal (iTerm3) **with colors!** But I have not managed to get this working. I use the same setup with `geth` (starting geth in a `start_geth.sh` script I've written, see below.), and it works for geth - works meaning the log contains the ANSI color codes explicitly written out, which means that when I `tail -f` my geth.log I see colors!

My `start_lighthouse_bn.sh` script looks something like this, and this fails to keep ANSI colors intact, as in the `$LIGHTHOUSE_BN_LOG` log file are stripped of ANSI color codes => no color with `tail`

```sh
#!/bin/zsh
source /Users/sajjon/Desktop/Services/Scripts/set_env.sh
source "lighthouse --network mainnet beacon_node --logfile $LIGHTHOUSE_BN_LOG"
```

I've also tried this, which gives same (bad) result, i.e. no colors.
```sh
#!/bin/zsh
source /Users/sajjon/Desktop/Services/Scripts/set_env.sh
cmd="lighthouse --network mainnet beacon_node"
script -Fq $LIGHTHOUSE_BN_LOG /bin/zsh -c "$cmd"
```

However, the analogous script I use for starting geth works and results in $GETH_LOG contain ANSI color codes => colors with `tail`.

```sh
#!/bin/zsh
source /Users/sajjon/Desktop/Services/Scripts/set_env.sh
cmd="geth"
script -Fq $GETH_LOG /bin/zsh -c "$cmd"
```

Ofc Lighthouse is written in Rust and Geth in Go, so not strange that they two different underlying loggers work differently, but I was hoping that I'm just doing something wronge - OR COMPLETELY OVERENGINEERING THINGS 😛 Do you have any quick fix for me?

Or is a code change in how Lighthouse loggers work needed?

An important note is that when I `~/.lighthouse/mainnet/beacon/logs/beacon.log` I get no colors either, and when I open them, they lack ASNI color codes. And when I open the geth log I produce with `script -Fq ...` above, it looks like this:

<img width="926" alt="Screenshot 2022-08-30 at 11 51 56" src="https://user-images.githubusercontent.com/864410/187406975-60ad053d-abc0-4fe3-a935-2a2156fd5fd3.png">

Another important note - which is obvious but I'm stating it anyway - when I start lighthouse directly in my terminal I get colors! It is just when the log is written to logfile that I do not. So it should be possible to get the ANSI color codes written to the log file too right?

## Version

3.0.0

## Present Behaviour

Unable to output logs with ANSI color codes intact to a log file.

## Expected Behaviour

I expect log files to not be stripped of coloring, which DRASTICALLY increases readability of them! I'm very color...centric, seeing green makes me happy, and I instantly react to the red in logs, or yellow, so it is very important for me to be able to see this in log files.

## Steps to resolve
Dunno, minor change in underlying logger? Or in config to logger? [On these lines the format is configured](https://github.com/sigp/lighthouse/blob/6f732986f1a42bec9f7ef7f57a143836f585a119/lighthouse/environment/src/lib.rs#L216-L219), maybe that can be changed? Which might require a new possible value to the [`log-format` arg](https://github.com/sigp/lighthouse/blob/bac7c3fa544495a257722aaad9cd8f72fee2f2b4/lighthouse/src/main.rs#L133-L139)

**Comment by michaelsproul** [comment]
I think this would require some changes to the logger. We would need to add a CLI flag like `--log-colour` (alias `--log-color`) to force outputting coloured logs to the terminal. I don't think we should add a flag to write colour directly to logfiles.

The `TermDecorator` that we construct here could be modified using [`force_color`](https://docs.rs/slog-term/latest/slog_term/struct.TermDecoratorBuilder.html#method.force_color) according to the value of the CLI flag.

https://github.com/sigp/lighthouse/blob/18c61a5e8be3e54226a86a69b96f8f4f7fd790e4/lighthouse/environment/src/lib.rs#L142-L148

----

As a workaround you can use `unbuffer` as described in this post: https://unix.stackexchange.com/questions/636455/what-is-a-general-way-to-trick-programs-into-thinking-theyre-not-being-piped

I can confirm that using `unbuffer` results in coloured logs being output to `tee` and on to `log.txt`:

```
$ unbuffer lighthouse bn --network goerli 2>&1 | tee -ia log.txt
```

**Comment by Sajjon** [comment]
@michaelsproul nice! I might give it a go if I find some free time this weekend.

Unfortunately the `unbuffer` and `tee` solution did not work in macOS (Ventura) (after having `brew install expect` which seems to be the package unbuffwe is part of). Any ideas on alternative flags for macOS?

**Comment by michaelsproul** [comment]
> Any ideas on alternative flags for macOS?

This page suggests `stdbuf` but it doesn't seem to work with `lighthouse` (probably because we use static linking): https://stackoverflow.com/questions/3465619/how-to-make-output-of-any-shell-command-unbuffered

**Comment by michaelsproul** [comment]
Resolved by #3538
